daemon/logger/splunk/splunk.go

```diff
@@ -491,20 +494,20 @@ func (l *splunkLogger) tryPostMessages(messages []*splunkMessage) error {
 	if l.gzipCompression {
 		req.Header.Set("Content-Encoding", "gzip")
 	}
-	res, err := l.client.Do(req)
+	resp, err := l.client.Do(req)
 	if err != nil {
 		return err
 	}
-	defer res.Body.Close()
-	if res.StatusCode != http.StatusOK {
-		var body []byte
-		body, err = ioutil.ReadAll(res.Body)
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		rdr := io.LimitReader(resp.Body, maxResponseSize)
+		body, err := ioutil.ReadAll(rdr)
 		if err != nil {
 			return err
 		}
-		return fmt.Errorf("%s: failed to send event - %s - %s", driverName, res.Status, body)
+		return fmt.Errorf("%s: failed to send event - %s - %s", driverName, resp.Status, string(body))
 	}
-	io.Copy(ioutil.Discard, res.Body)
+	pools.Copy(ioutil.Discard, resp.Body)
```

Comment:
The way I read this is that, we want to read the resp.Body to completion into `ioutil.Discard`, so that the HTTP client connection gets reused. Yes? In that case, we need it in the case of non-OK status codes as well.